src/NewTools-Debugger/StDebuggerConfigurationToolbarCommandTreeBuilder.class.st

```diff
@@ -0,0 +1,46 @@
+Class {
+	#name : #StDebuggerConfigurationToolbarCommandTreeBuilder,
+	#superclass : #StDebuggerCommandTreeBuilder,
+	#category : #'NewTools-Debugger-Model'
+}
+
+{ #category : #accessing }
+StDebuggerConfigurationToolbarCommandTreeBuilder class >> displayStrategy [
+	^CmUIDisplayAsGroup
+]
+
+{ #category : #accessing }
+StDebuggerConfigurationToolbarCommandTreeBuilder class >> groupDescription [
+	^ 'Debugger configuration actions'
+]
+
+{ #category : #accessing }
+StDebuggerConfigurationToolbarCommandTreeBuilder class >> groupName [ 	
+	^'Configuration'
+]
+
+{ #category : #building }
+StDebuggerConfigurationToolbarCommandTreeBuilder >> buildDebuggerCommandGroup [
+
+	| group |
+	self group.
+	group := self group.
+	self configurationCommands do: [ :specCommand |
+		group register: specCommand ].
+	^ group
+]
+
+{ #category : #'accessing - commands classes' }
+StDebuggerConfigurationToolbarCommandTreeBuilder >> configurationCommandClasses [
+
+	^ {
+		  StDebuggerToggleFilterStackCommand.
+		  StDebuggerSettingsCommand }
```

Comment:
The classes shouldn't be hardcoded. So, to extend the debugger command classes, we need to change this collection? 

Instead of changing the code of the debugger, we would need to change the collection in this method. It just moves the problem elsewhere no?

Why don't we use a trait (or an abstract class) to make this method return `TStDebuggerConfigurationCommand users` (or `StDebuggerConfigurationCommand subclasses`)?

Reply:
Ok, I just realized we can use `<extensionCommands>` in `StDebugger class`